charts/visor/templates/deployment.yaml

```diff
@@ -65,28 +76,95 @@ spec:
       - name: visor
         image: "{{ .Values.image.repo }}:{{ .Values.image.tag }}"
         imagePullPolicy: "{{ .Values.image.pullPolicy }}"
-        args: ["--tracing={{ .Values.jaeger.enabled }}", "run"]
+        args:
+          # Global args
+          # debugging args
+          {{- if and .Values.jaeger .Values.jaeger.enabled }}
+          - "--tracing={{ .Values.jaeger.enabled }}"
+          - "--jaeger-agent-host={{ .Values.jaeger.host }}"
+          - "--jaeger-agent-port={{ .Values.jaeger.port }}"
+          - "--jaeger-service-name={{ .Values.jaeger.serviceName }}"
+          {{- if .Values.jaeger.sampler }}
+          - "--jaeger-sampler-type={{ .Values.jaeger.sampler.type }}"
+          - "--jaeger-sample-param={{ .Values.jaeger.sampler.param }}"
+          {{- end }}
+          {{- end }}
+
+          {{- if .Values.logLevel }}
+          - "--log-level={{ .Values.logLevel }}"
+          {{- end }}
+
+          {{- if .Values.logLevelNamed }}
+          - "--log-level-named={{ .Values.logLevelNamed }}"
+          {{- end }}
+
+          # database args
+          {{- if .Values.pgPoolSize }}
+          - "--db-pool-size={{ int .Values.pgPoolSize }}"
+          {{- end }}
+
+          {{- if .Values.pgAppName }}
+          - "--name={{ .Values.pgAppName }}"
+          {{- end }}
+
+          - "--allow-schema-migration={{ .Values.allowAutomaticMigration }}"
+
+          # lens config
+          {{- if .Values.lens }}
+          {{- if and .Values.lens.lotusAPI }}
+          - "--lens=lotus"
+          {{- $lotusAPIMultiaddr := "/dns/{{ .Release.Name }}-lotus-daemon.{{ .Release.Namespace }}.svc.cluster.local/tcp/1234" -}}
+          {{- if .Values.lens.lotusAPI.lotusAPIMultiaddr -}}
+            {{- $lotusAPIMultiaddr := .Values.lens.lotusAPI.lotusAPIMultiaddr -}}
+          {{- end }}
+          - "--repo=$(LOTUS_API_TOKEN):{{ $lotusAPIMultiaddr }}"
```

Comment:
Think this should be --api

Reply:
Also, I'd like to split --api into two separate flags in visor: --api and --api-token. Makes the separation cleaner and means the token can very easily come from a secret without munging into a composite value.

Reply:
Let's include this in the next release.

Reply:
Ended up walking this change back for now and will continue the existing env-var approach.